.github/workflows/ci.yml

```diff
@@ -38,16 +37,18 @@ jobs:
         - { path: omnibus, name: omnibus }
         - { path: python, name: python }
         - { path: terraform, name: terraform }
+    env:
+      BASE_IMAGE: ubuntu:18.04
+      CORE_IMAGE: dependabot/dependabot-core
+      CORE_BRANCH_IMAGE: ghcr.io/dependabot/dependabot-core-branch
+      CORE_CI_IMAGE: dependabot/dependabot-core-ci
+      CORE_CI_MIRROR: ghcr.io/dependabot/dependabot-core-ci
+      CODE_DIR: /home/dependabot/dependabot-core
     steps:
       - name: Checkout code
         uses: actions/checkout@v2
-      - name: Prepare environment variables
-        run: |
-          echo "BRANCH_REF=$(echo '${{ github.ref }}' | sed -E 's/[^A-Za-z0-9]+/-/g')" >> $GITHUB_ENV
-          echo "BASE_IMAGE=ubuntu:18.04" >> $GITHUB_ENV
-          echo "CORE_IMAGE=dependabot/dependabot-core" >> $GITHUB_ENV
-          echo "CORE_CI_IMAGE=dependabot/dependabot-core-ci" >> $GITHUB_ENV
-          echo "CODE_DIR=/home/dependabot/dependabot-core" >> $GITHUB_ENV
+      - name: Prepare BRANCH_REF environment variable
+        run: echo "BRANCH_REF=$(echo '${{ github.ref }}' | sed -E 's/[^A-Za-z0-9]+/-/g')" >> $GITHUB_ENV
```

Comment:
Taking another look at this, are we accounting for the possibility of multiple PRs coming from different forks with the same downstream branch name? Looking at [the docs](https://docs.github.com/en/actions/learn-github-actions/contexts), it's not obvious to me that `github.ref` captures this...

Reply:
Oh wait, never mind. Please disregard my last comment. `github.ref` uses the PR number (not branch name), which is indeed unique.

Reply:
I noticed that a [fork of Dependabot-Core](https://github.com/dependabot/dependabot-core/runs/4644267200?check_suite_focus=true#step:6:118) is _correctly_ not able to push its branch but the error message is unclear on how to fix this.

I'm not sure whether this error is produced from a failing to push to `dockerhub` or `ghcr`, so maybe we should be more explicit about disallowing forked repo/branch CI runs and exit early.

Reply:
Resolved by https://github.com/dependabot/dependabot-core/pull/4569